src/test/kotlin/uk/gov/dluhc/printapi/rds/repository/CertificateRepositoryTest.kt

```diff
@@ -0,0 +1,108 @@
+import uk.gov.dluhc.printapi.testsupport.testdata.aValidVacNumber
+import uk.gov.dluhc.printapi.testsupport.testdata.aValidVacVersion
+import uk.gov.dluhc.printapi.testsupport.testdata.aValidWebsite
+import uk.gov.dluhc.printapi.testsupport.testdata.zip.aPhotoArn
+
+internal class CertificateRepositoryTest : IntegrationTest() {
+
+    @Test
+    fun `should return Certificate given persisted Certificate with all required properties`() {
+        // Given
+        val certificate = Certificate(
+            vacNumber = aValidVacNumber(),
+            sourceType = aValidSourceType(),
+            sourceReference = aValidSourceReference(),
+            applicationReference = aValidApplicationReference(),
+            applicationReceivedDateTime = aValidApplicationReceivedDateTime(),
+            issuingAuthority = aValidIssuingAuthority(),
+            issueDate = aValidIssueDate(),
+            suggestedExpiryDate = aValidSuggestedExpiryDate(),
+            gssCode = aGssCode(),
+            status = aValidCertificateStatus()
+        )
+        val deliveryAddress = Address(
+            street = aValidAddressStreet(),
+            postcode = aValidAddressPostcode()
+        )
+        val delivery = Delivery(
+            addressee = aValidDeliveryName(),
+            address = deliveryAddress,
+            deliveryClass = aValidDeliveryClass(),
+            deliveryMethod = aValidDeliveryMethod()
+        )
+        val eroEnglish = ElectoralRegistrationOffice(
+            address = Address(
+                street = aValidAddressStreet(),
+                postcode = aValidAddressPostcode()
+            ),
+            name = aValidEroName(),
+            phoneNumber = aValidPhoneNumber(),
+            emailAddress = aValidEmailAddress(),
+            website = aValidWebsite()
+        )
+        val printRequest = PrintRequest(
+            requestId = aValidRequestId(),
+            vacVersion = aValidVacVersion(),
+            requestDateTime = aValidRequestDateTime(),
+            firstName = aValidFirstName(),
+            surname = aValidSurname(),
+            certificateLanguage = aValidCertificateLanguage(),
+            certificateFormat = aValidCertificateFormat(),
+            photoLocationArn = aPhotoArn(),
+            delivery = delivery,
+            eroEnglish = eroEnglish,
+            eroWelsh = null,
+            userId = aValidUserId()
+        )
+        val printRequestStatus = PrintRequestStatus(
+            status = aValidCertificateStatus(),
+            eventDateTime = aValidPrintRequestStatusEventDateTime(),
+        )
+        printRequest.addPrintRequestStatus(printRequestStatus)
+        certificate.addPrintRequest(printRequest)
+        val expected = certificateRepository.save(certificate)
+
+        // When
+        val actual = certificateRepository.findById(expected.id!!)
+
+        // Then
+        assertThat(actual).isPresent
+        assertThat(actual.get()).isEqualTo(expected)
```

Comment:
@Neil-Massey, this will use the equals method of the Certificate, which only checks the equality of the id. To check if expected and actual are equal for all the fields we need to use recursive comparison.

<pre>
assertThat(actual.get()).<b>usingRecursiveComparison()</b>.isEqualTo(expected)
</pre>